Separate configurations for colorized and json logs in Task SDK / Celery Executor

highlyavailable opened the pull request:
Fixes #50963

This PR separates the control of log format (console vs JSON) from log styling (colors and rich traces) in the Task SDK logging configuration. Previously, the single `enable_pretty_log` boolean controlled both aspects, making it impossible to have console logs without colors and rich traces.

## Problem
- Single enable_pretty_log boolean controlled BOTH format (console vs JSON) AND styling (colors + rich traces)
- No way to get console logger without colors and rich traces
- Celery workers showed color escape sequences as raw text in log ingestion solutions like Splunk

## Solution
- Separated enable_pretty_log (format) from enable_colors (styling)
- enable_pretty_log=True + enable_colors=False = plain text console logger (no colors, no rich traces)
- Respects existing colored_console_log configuration parameter
- Celery workers automatically respect the configuration without code changes
- Change is backwards compatible since:
  - `configure_logging()` - uses default `enable_colors=None` → reads config
  - `logging_processors(enable_pretty_log=True)` - uses default `enable_colors=True`

### New Behavior Matrix:
| enable_pretty_log | enable_colors | Result |
|-------------------|----------------|---------|
| True | True | Colorized console + rich traces + "%Y-%m-%d %H:%M:%S.%f" timestamps |
| True | False | Plain text console + no rich traces + "%Y-%m-%d %H:%M:%S.%f" timestamps |
| False | True/False | JSON logger + ISO timestamps + dict exception renderer |

## Testing
All new & existing tests are passing, as well as the pre-commit checks.  I wrote some simple unit tests in the test SDK, 
- `test_logging_processors_with_colors()`: Verifies colored console renderer
- `test_logging_processors_without_colors()`: Verifies non-colored console renderer
- `test_logging_processors_json_format()`: Verifies JSON renderer unchanged
- `test_configure_logging_respects_colored_console_log_config()`: Tests configuration reading
- `test_configure_logging_explicit_enable_colors()`: Tests explicit parameter override
- `test_configure_logging_no_airflow_config()`: Tests fallback behavior

Since I am new to the project, I also wrote a short testing script (attached as a `.txt`) and ran it in my local Breeze environment, here were the results:

[demo_test.txt](https://github.com/user-attachments/files/20443592/demo_test.txt)

<img width="751" alt="image" src="https://github.com/user-attachments/assets/4618c04f-04f9-4bbd-8dd1-8f4d3c647634" />

highlyavailable committed:
Tests for new log configuration

highlyavailable committed:
Enhance logging configuration by adding 'enable_colors' parameter to 'logging_processors' and 'configure_logging' functions. This allows for customizable console output with or without colors based on user preference or configuration settings.

uranusjr commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -176,20 +176,32 @@ def logging_processors(enable_pretty_log: bool, mask_secrets: bool = True):
+            rich_exc_formatter = structlog.dev.RichTracebackFormatter(
+                suppress=suppress,
+                show_locals=False,
+            )
```
Why does `enable_colors=False` affect formatting options not related to colors?

uranusjr commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -261,13 +274,24 @@ def configure_logging(
 
+    # If enable_colors is not explicitly set, read from configuration
+    if enable_colors is None:
+        if "airflow.configuration" in sys.modules:
```
Why do we need to check this?

amoghrajesh replied:
+1 same qn, we should just be able to do:

```
            from airflow.configuration import conf

            enable_colors = conf.getboolean("logging", "colored_console_log", fallback=True)
```

amoghrajesh replied:
Its handled now

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -140,7 +140,7 @@ def emit(self, record: logging.LogRecord):
 
 @cache
-def logging_processors(enable_pretty_log: bool, mask_secrets: bool = True):
+def logging_processors(enable_pretty_log: bool, mask_secrets: bool = True, enable_colors: bool = True):
```
Can we just use the same name as the config? 
https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#colored-console-log

amoghrajesh replied:
Been handled

highlyavailable committed:
Address review comments: use direct imports and match config naming

- Remove conditional imports, use direct \
- Rename parameter from \ to \ to match existing config
- Remove unrelated formatting options (show_locals) from color logic
- Update tests to reflect the changes

vatsrahul1001 committed:
Merge branch 'main' into fix/50963_separate_colorized_json_logs

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -252,22 +266,30 @@ def configure_logging(
-            log_level = conf.get("logging", "logging_level", fallback="INFO")
+    # If colored_console_log is not explicitly set, read from configuration
+    if colored_console_log is None:
+        from airflow.configuration import conf
```
Non blocking comment, move the import up in the function to reuse

amoghrajesh reviewed (approved):
One comment, more like a nit, can be addressed now or in a followup.
